IanGRap/Squish
# Squish Detection

Contributors: Austin

Goal:
Implement player death when they get squished by a block

Important Files Changed:
- SquishDetector.cs

Changes:
- Added a child object to the player prefab called SqishDetector
    - It is just a trigger sphere collider with a script attached 
    - SquishDetector.cs: Keeps track of all blocks currently overlapping with the player object and destroys 
      the player if they remain overlapped with a block for too long (~0.1 seconds is the sweetspot, I think)

Customizable Fields within Unity:
- Squish Sensitivity: This determines the scale of the Squish Detection collider. When the number is larger, it will be easier to get squished by the corner of a block. When the number is low, if the corner of a block collides with the player, they will likely be pushed out of the way instead of killed. Crank up the sensitivity to make the game more punishing
- Squished Buffer Time: This value probably won't need to be changed. While I was testing, I noticed that sometimes, Unity's physics would cause the player to dash into the blocks for about a frame. This was causing unintended squishing, so I added a buffer time. Right now, the block must be overlapping with the squish detector for 0.1 seconds in order for a squish to occur. If you want to see why this field is needed, try setting it to 0 and dashing into a few blocks (you will likely die).

How to test:
I created 2 scenes inside Assets/Scenes/Squish Detection. Try getting squished in both:

1) Control the Player: Here, you can control the player as blocks are falling from above. You can get squished only from above in this scene.
2) Control the Block: Here, there are 4 player objects, all with keyboard input toggled off. There is also 1 block with keyboard input toggled on. In this scene, you can test squishing the player from above, blow, left and right. 

After testing basic squishing, modify Squish Sensitivity and try to get squished by just the corner of blocks. I think a decent sensitivity is 0.75 but it should be customizable.
